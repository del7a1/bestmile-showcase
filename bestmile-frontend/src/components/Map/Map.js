import React, { Component } from 'react';
import L from "leaflet";
import 'leaflet-routing-machine';
import SockJS from "sockjs-client"
import Stomp from "stompjs"

const style = {
  width: "100%",
  height: "500px"
};

const newYorkCoordinates = [40.709465026855469,-73.797340393066406];

class Map extends Component {

  componentDidMount() {
    // create map
    this.map = L.map("map", {
      center: newYorkCoordinates,
      zoom: 12,
      minZoom: 12,
      layers: [
        L.tileLayer("http://{s}.tile.osm.org/{z}/{x}/{y}.png", {
          attribution:
            '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
        })
      ]
    });

    let socket = new SockJS('http://localhost:8080/subscribe');
    let stompClient = Stomp.over(socket);
    stompClient.connect({}, (frame) => {

      stompClient.subscribe('/user/mission/all', (msg) => {
        JSON.parse(msg.body).forEach((mission) => this.addMission(mission));
      });

      stompClient.subscribe('/mission/new', (msg) => {

        this.addMission(JSON.parse(msg.body));
      });
    });

    this.map.on('moveend ', (event) => {
       console.log(this.map.getBounds());
       stompClient.send("/app/mission", {}, this.map.getBounds());
     });

  }

  addMission(mission) {
    let myIcon = L.icon({iconUrl: 'taxi-logo.png', iconSize: [25, 25]});
    L.marker(mission.currentPositionCoordinate, {icon: myIcon})
      .bindPopup('<p>Hello world!<br />This is a nice popup.</p>')
      .openPopup()
      .addTo(this.map);
    let polyline = L.polyline(mission.route.waypoints, {color: 'blue'}).addTo(this.map);
  }

  render() {
    return (
      <div>
        <div id="map" style={style} />;
      </div>
    );
  }

}

export default Map;
