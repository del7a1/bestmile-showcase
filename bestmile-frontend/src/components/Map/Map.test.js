import React from 'react';
import Map from './Map';

describe('Map', () => {

 it('Should be defined', () => {
   expect(Map).toBeDefined();
 });

 it('Should render correctly', () => {
   const container = document.createElement('div');
   document.body.appendChild(container);
   const wrapper = mount(<Map />, { attachTo: container });

   expect(wrapper).toMatchSnapshot();
 });

});
