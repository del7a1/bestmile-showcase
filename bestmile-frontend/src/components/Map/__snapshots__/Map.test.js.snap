// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Should render correctly 1`] = `
<Map>
  <div>
    <div
      id="map"
      style={
        Object {
          "height": "500px",
          "width": "100%",
        }
      }
    />
  </div>
</Map>
`;
